src/server/tran_server.cpp

```diff
@@ -452,21 +443,81 @@ tran_server::connection_handler::connection_handler (cubcomm::channel &&chn, tra
   // Transaction server will use message specific error handlers.
   // Implementation will assert that an error handler is present if needed.
 
-  m_conn.reset (new page_server_conn_t (std::move (chn), std::move (request_handlers), tran_to_page_request::RESPOND,
+  auto lockg = std::lock_guard<std::shared_mutex> { m_state_mtx };
+  assert (m_state == state::OPEN);
+
+  m_state = state::CONNECTING;
```

Comment:
The transition from CONNECTING to CONNECTED is very short. Separating the states means increasing the complexity, so why not combine the two into one state? If you dis-agree, please explain a bit more about the benefits of splitting the two.

Reply:
Now, yes, the CONNECTING is not open to the outside. It's just defining the state and the `CONNECTING` will be used in the near future when the catch-up process comes in. We can add the state then, but I've just added it while I'm at it.

Reply:
I would leave the `CONNECTING` state. That was the idea -  to have a clear image of what state is the handler is.